network/kademlia.go

```diff
@@ -203,27 +203,56 @@ func (k *Kademlia) removeFromCapabilityIndex(p interface{}, disconnectOnly bool)
 	}
 }
 
+const (
+	black = iota
+	red
```

Comment:
Because after all we don't "use" whites? :)

Reply:
Yes, I was inspired by black-red trees ([https://en.wikipedia.org/wiki/Red%E2%80%93black_tree](url). It is not exactly the same but we are too using node colurs to try to get some balance.